### PR TITLE
Sequence heatmap implementation

### PR DESCRIPTION
**Pull request by marcelo-q**

### CODE REVIEW

**Comment by gustavo-salazar** [comment]
@marcelo-q the tests are failing, but from what I can see, it's just missing the snapshot files, do you mind including those?

**Comment by marcelo-q** [comment]
hi @gustavo-salazar, thanks for the review.

sorry about that, just added the snapshot files generated by test now

**Comment by gustavo-salazar** [comment]
> hi @gustavo-salazar, thanks for the review.
> 
> sorry about that, just added the snapshot files generated by test now

Hey no prob, unfortunately it is [still failing](https://github.com/ebi-webcomponents/nightingale/actions/runs/9000083251/job/24725089157), I think there is still one more snaphsot missing

**Comment by marcelo-q** [comment]
I just removed a not needed test file that was placeholder before we renamed the lib. Maybe that was the fix

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (gustavo-salazar) [inline packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts:413]: The highlight attribute could have multiple segments, for instance: `highlight="20:25,30:35"`. 
Have a look to `this.highlightedRegion.segments` where the multiple segments are already parsed. That should be available already here as you are implementing `withHighlight.

File: stories/15.SequenceHeatmap/NightingaleSequenceHeatmap.stories.ts
Comment (gustavo-salazar) [inline stories/15.SequenceHeatmap/NightingaleSequenceHeatmap.stories.ts:40]: Maybe import these values from your mockup data to simplify the code of the story.

**Review by gustavo-salazar — CHANGES_REQUESTED** [review]
Hey @marcelo-q, thanks so much for this PR.

Everything works as expected. You can take most my comments as suggestions. The only one that really needs to be changed before merging is the support of multiple highlight segments.

As an extra thing, can you add the component to the Manager story? that way we can see it playing along with other Nightingale components. And will also test if the `withResizable` is doing what is expected of it.

Another comment is if you had a look to the colored sequence component. It seems to me that the functionality is quite similar, with the major difference being that this one is implemented on SVG and yours is on canvas. Which in itself is reason enough to have another component. I'm just wondering if that's the only reason to create a new component instead of reusing the one that was there.

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (marcelo-q) [reply packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts]: Thanks this is solved now

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (marcelo-q) [reply packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts]: This is also fixed now. Took me a bit of work since I did not notice applyZoomTranslation also needed to be overriden since it's d3 / SVG coupled

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (marcelo-q) [reply packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts]: Fixed as reply above

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (marcelo-q) [reply packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts]: Nice catch, thanks. This was related to the heatmap lib wrapping. Now fixed as I added requestUpdate alongside these conditionals

File: packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts
Comment (marcelo-q) [reply packages/nightingale-sequence-heatmap/src/nightingale-sequence-heatmap.ts]: Refactored so it's fixed, thanks.

File: stories/15.SequenceHeatmap/NightingaleSequenceHeatmap.stories.ts
Comment (marcelo-q) [reply stories/15.SequenceHeatmap/NightingaleSequenceHeatmap.stories.ts]: Fixed

**Comment by marcelo-q** [comment]
> Hey @marcelo-q, thanks so much for this PR.
> 
> Everything works as expected. You can take most my comments as suggestions. The only one that really needs to be changed before merging is the support of multiple highlight segments.
> 
> As an extra thing, can you add the component to the Manager story? that way we can see it playing along with other Nightingale components. And will also test if the `withResizable` is doing what is expected of it.
> 
> Another comment is if you had a look to the colored sequence component. It seems to me that the functionality is quite similar, with the major difference being that this one is implemented on SVG and yours is on canvas. Which in itself is reason enough to have another component. I'm just wondering if that's the only reason to create a new component instead of reusing the one that was there.

Hi @gustavo-salazar thank you very much for taking your time to review this

I've changed my code according to your comments so it fits the Nightingale lifecycle better (a bit embarrassed I did not do this before haha). If you have spotted anything else feel free to add more comments and I will be happy to try to solve it.

I also added the component to the Manager story.

Lastly I believe our main reasoning for doing this is that we are generating heatmaps with very large amounts of data in our different pages. There were some performance issues with SVG before that prompted Sameer and Misi (previous line manager) to push for adopting canvas based visualisations in our group.

To be honest Sreenath and Sameer would be able to update you better on this. Maybe we can start a more detailed discussion on slack

**Comment by gustavo-salazar** [comment]
> Hi @gustavo-salazar thank you very much for taking your time to review this
> 
> I've changed my code according to your comments so it fits the Nightingale lifecycle better (a bit embarrassed I did not do this before haha). If you have spotted anything else feel free to add more comments and I will be happy to try to solve it.
>

Awesome. Just have a quick look and all seems to be OK.

> I also added the component to the Manager story.
> 

I was going to tell you that you forgot to import the component in the story. However, the story works well without it, and on a quick test it doesn't seem to need all the imports that we have at the top of that file, which then makes me wonder, how is that working, where are those elements being defined? is it something storybook does? anyways not your problem, but if you have any clue about it please share,

> Lastly I believe our main reasoning for doing this is that we are generating heatmaps with very large amounts of data in our different pages. There were some performance issues with SVG before that prompted Sameer and Misi (previous line manager) to push for adopting canvas based visualisations in our group.
> 
> To be honest Sreenath and Sameer would be able to update you better on this. Maybe we can start a more detailed discussion on slack

Yeah, very similar reasoning to why we used canvas on the multiple alignment component. No need at the moment to have a more detailed conversation, as I was just curious. But good to know who to ask about it.

**Review by gustavo-salazar — APPROVED** [review]

**Comment by gustavo-salazar** [comment]
I just approved this PR. I will give it a few days in case any of the UniProt fellas have the time to do a second review. Otherwise, I'll merge and publish the components next week, maybe together with the variation track, and hopefully with a dependencies update!

**Review by Swaathik — APPROVED** [review]
Thank you @marcelo-q and @gustavo-salazar, we are fine with the changes and for the merging of the PR

**Comment by gustavo-salazar** [comment]
I'm merging now. But will only publish the components next week, together with the `variation` ones

**Pull request merged**
